File: leet_1351_count_negative.py
class Solution:
    def countNegatives(self, g: List[List[int]]) -> int:
        R = len(g)
        C = len(g[0])
        r = 0
        c = C - 1
        res = 0
        while r < R and c > -1:
            if g[r][c] < 0:
                res += R - r
                c -= 1
            else:
                r += 1
        return res

File: leet_1351_count_negative.go
func countNegatives(g [][]int) int {
    r := len(g)
    c := len(g[0])
    i := 0
    j := c - 1
    res := 0
    for i < r && j > -1 {
        if g[i][j] < 0 {
            res += r - i
            j--
        } else {
            i++
        }
    }
    return res

}
